t/t2000-imp-exec.t

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+#
+
+test_description='IMP exec basic functionality test
+
+Basic flux-imp exec functionality and corner case handing tests
+'
+
+# Append --logfile option if FLUX_TESTS_LOGFILE is set in environment:
+test -n "$FLUX_TESTS_LOGFILE" && set -- "$@" --logfile
+. `dirname $0`/sharness.sh
+
+flux_imp=${SHARNESS_BUILD_DIRECTORY}/src/imp/flux-imp
+sign=${SHARNESS_BUILD_DIRECTORY}/t/src/sign
+
+echo "# Using ${flux_imp}"
+
+fake_imp_input() {
+	printf '{"J":"%s"}' $(echo $1 | $sign)
+}
+
+test_expect_success 'flux-imp exec returns error when run with no args' '
+	test_must_fail $flux_imp exec
+'
+test_expect_success 'flux-imp exec returns error when run with only 1 arg' '
+	test_must_fail $flux_imp exec foo
+'
+test_expect_success 'flux-imp exec returns error with no input' '
+	test_must_fail $flux_imp exec shell arg < /dev/null
+'
+test_expect_success 'flux-imp exec returns error with bad input ' '
+	echo foo | test_must_fail $flux_imp exec shell arg
+'
+test_expect_success 'flux-imp exec returns error with bad JSON input ' '
+	echo "{}" | test_must_fail $flux_imp exec shell arg
+'
+test_expect_success 'flux-imp exec returns error with bad JSON input ' '
+	echo "{}" | test_must_fail $flux_imp exec shell arg
+'
+test_expect_success 'create configs for flux-imp exec and signer' '
+	cat <<-EOF >sign-none.toml
+	allow-sudo = true
+	[sign]
+	max-ttl = 30
+	default-type = "none"
+	allowed-types = [ "none" ]
+	[exec]
+	allowed-users = [ "$(whoami)" ]
+	allowed-shells = [ "id", "echo" ]
+	EOF
+	cat <<-EOF >sign-none-allowed-munge.toml
+	allow-sudo = true
+	[sign]
+	max-ttl = 30
+	default-type = "none"
+	allowed-types = [ "munge" ]
+	[exec]
+	allowed-users = [ "$(whoami)" ]
+	allowed-shells = [ "id", "echo" ]
+	EOF
+'
+test_expect_success 'flux-imp exec works in unprivileged mode' '
+	( export FLUX_IMP_CONFIG_PATTERN=sign-none.toml  &&
+	  fake_imp_input foo | $flux_imp exec echo good >works.out
+	) &&
+	cat >works.expected <<-EOF &&
+	good
+	EOF
+	test_cmp works.expected works.out
+'
+test_expect_success 'flux-imp exec fails when signature type not allowed' '
+	( export FLUX_IMP_CONFIG_PATTERN=./sign-none-allowed-munge.toml  &&
+	  fake_imp_input foo | \
+	    test_must_fail $flux_imp exec echo good >badmech.log 2>&1
+	) &&
+	test_debug "cat badmech.log" &&
+	grep "signature validation failed" badmech.log
+'
+test_expect_success 'flux-imp bails out with no configuration' '
+	( export FLUX_IMP_CONFIG_PATTERN=/nosuchthing &&
+	  fake_imp_input foo | \
+	    test_must_fail $flux_imp exec echo good >badconf.log 2>&1
+	) &&
+	grep -i "Failed to load configuration" badconf.log
+'
+test_expect_success 'flux-imp exec fails with invalid stdin input' '
+	( export FLUX_IMP_CONFIG_PATTERN=sign-none.toml &&
+	  echo foo | \
+	    test_must_fail $flux_imp exec echo good >badjson.log 2>&1
+	)
```

Comment:
broken && chain?   (Same for tests following this one)

Reply:
Yeah, good catch, thanks! These were all cut-and-paste so I'll fix em all